src/logger.ts

```diff
@@ -53,16 +56,16 @@ export class Logger {
   static DEFAULT_OPTIONS:
       Readonly<LoggerConfig> = {level: 'error', levels: LEVELS, tag: ''};
 
-  // TODO: Mark this private when TypeScript 2.9 comes out.
-  // See https://github.com/Microsoft/TypeScript/issues/20080 for more
-  // information.
-  [kTag]: string;
+  private[kTag]: string;
 
-  /**
-   * Emits a log at this log level.
-   */
-  // tslint:disable-next-line:no-any
-  [logLevel: string]: (...args: any[]) => this;
+  // tslint:disable:no-any
+  silent!: (...args: any[]) => this;
```

Comment:
Why do you need the non-null annotations in these fields?

Reply:
The TypeScript compiler doesn't know that these will be set for default options. I'll add a note in the comments.